jdftx/scripts/ase/JDFTx.py

```diff
@@ -164,6 +202,11 @@ def constructInput(self, atoms):
 			inputfile += 'ion %s %f %f %f \t 1\n' % (atomNames[i], atomPos[i][0], atomPos[i][1], atomPos[i][2])
 		del i
 
+		# Add k-points
+		if self.kpoints:
+			for pt in self.kpoints:
+				inputfile += 'kpoint %.8f %.8f %.8f %.14f\n' % pt
```

Comment:
jdftx checks for sum of weights to be 1 +/- 1e-12, so weight format needs extra precision